c-api-examples/offline-stt-c-api.c

```diff
@@ -0,0 +1,98 @@
+// c-api-examples/offline-stt-c-api.c
+//
+// Copyright (c)  2024  Xiaomi Corporation
+
+// We assume you have pre-downloaded the whisper multi-lingual models
+// from https://github.com/k2-fsa/sherpa-onnx/releases/tag/asr-models
+// An example command to download the "tiny" whisper model is given below:
+//
+// clang-format off
+//
+// wget https://github.com/k2-fsa/sherpa-onnx/releases/download/asr-models/sherpa-onnx-whisper-tiny.tar.bz2
+// tar xvf sherpa-onnx-whisper-tiny.tar.bz2
+// rm sherpa-onnx-whisper-tiny.tar.bz2
+//
+// clang-format on
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "sherpa-onnx/c-api/c-api.h"
+
+int32_t main() {
+  // You can find more test waves from
+  // https://hf-mirror.com/spaces/k2-fsa/spoken-language-identification/tree/main/test_wavs
+  const char *wav_filename = "./sherpa-onnx-whisper-tiny/test_wavs/0.wav";
+  const char *encoder_filename = "sherpa-onnx-whisper-tiny/tiny-encoder.onnx";
+  const char *decoder_filename = "sherpa-onnx-whisper-tiny/tiny-decoder.onnx";
+  const char *tokens_filename = "sherpa-onnx-whisper-tiny/tiny-tokens.txt";
+  const char *language = "en";
+  const char *provider = "cpu";
+
+  const SherpaOnnxWave *wave = SherpaOnnxReadWave(wav_filename);
+  if (wave == NULL) {
+    fprintf(stderr, "Failed to read %s\n", wav_filename);
+    return -1;
+  }
+
+  // Whisper config
+  SherpaOnnxOfflineWhisperModelConfig whisper_config;
+  whisper_config.decoder = decoder_filename;
+  whisper_config.encoder = encoder_filename;
+  whisper_config.language = language;
+  whisper_config.tail_paddings = 0;
+  whisper_config.task = "transcribe";
+
+  // Offline model config
+  SherpaOnnxOfflineModelConfig offline_model_config;
```

Comment:
Could you use `memset` to zero-intiialize the struct?

Reply:
Added